### PR TITLE
add an e2e build target for ios

### PR DESCRIPTION
**Pull request by jakubgs**
State: DONE :muscle:  

Beginnings of setup for building of iOS app for end-to-end tests. Resolves: https://github.com/status-im/status-react/issues/6682

### CODE REVIEW

**Comment by jakubgs** [comment]
IT'S ALIVE:
https://ci.status.im/job/status-react/job/pull%20requests/job/PR-6785/70/

<blockquote><div><strong><a href="https://ci.status.im/job/status-react/job/pull%20requests/job/PR-6785/70/">status-react » pull requests » PR-6785 #70 [Jenkins]</a></strong></div></blockquote>

**Comment by hetvart** [comment]
> IT'S ALIVE:
> https://ci.status.im/job/status-react/job/pull%20requests/job/PR-6785/70/
> 
> > **[status-react » pull requests » PR-6785 #70 [Jenkins]](https://ci.status.im/job/status-react/job/pull%20requests/job/PR-6785/70/)**

I tried the build on local ios simulator and did not succeed to run it.
@mandrigin , could you please assist here?
maybe, wrong build opts: https://github.com/status-im/status-react/pull/6785/files#diff-56c5021ef06ac6cea807ae53269b6f4fR84 ?

<blockquote><div><strong><a href="https://ci.status.im/job/status-react/job/pull%20requests/job/PR-6785/70/">status-react » pull requests » PR-6785 #70 [Jenkins]</a></strong></div></blockquote>
<blockquote><div> GitHub</div><div><strong><a href="https://github.com/status-im/status-react/pull/6785">add an e2e build target for ios by jakubgs · Pull Request #6785 · status-im/status-react</a></strong></div><div>State: WIP
Beginnings of setup for building of iOS app for end-to-end tests. Resolves: #6682</div></blockquote>

**Comment by mandrigin** [comment]
@hetvart should it even run locally if you build it on Jenkins? Saucelabs documented that they do some certificate-black-magic to run stuff on their end...

**Comment by hetvart** [comment]
> @hetvart should it even run locally if you build it on Jenkins? Saucelabs documented that they do some certificate-black-magic to run stuff on their end...

@mandrigin 
not sure about this certification thing on simulators. anyway, this `*.app` brings the same result on both local and saucelab VMs - cannot run it. 

https://app.saucelabs.com/tests/59560ce14ba1446fa1bc678485eca1b7/watch#0

<blockquote><div><strong><a href="https://app.saucelabs.com/tests/59560ce14ba1446fa1bc678485eca1b7/watch#0">Sauce Labs | Dashboard</a></strong></div></blockquote>

File: fastlane/Fastfile [hunk 1]
```diff
@@ -71,6 +71,37 @@ def build_ios_adhoc
...
+
+    xcarchive_path = build_app(
+        sdk: "iphoneos",
+        scheme: "StatusIm",
+        workspace: "ios/StatusIm.xcworkspace",
+        configuration: "Release",
+        export_method: "development",
+        build_path: 'status-e2e',
+        output_directory: 'status-e2e',
+        output_name: 'StatusIm.app',
```
Comment (mandrigin) [inline fastlane/Fastfile]: @jakubgs can you also add
```
xcargs: "-sdk 'iphonesimulator11.3' -destination 'generic/platform=iOS Simulator'"
```

here? so it builds for the simulator

Comment (jakubgs) [reply fastlane/Fastfile]: Thanks man, will try.

**Comment by mandrigin** [comment]
@jakubgs I have fastlane installed, let me try to debug it locally

**Comment by mandrigin** [comment]
otherwise it will take forever to iterate

**Comment by jakubgs** [comment]
Doesn't seem like Fastlane likes this `sdk` setting:
```
#error architecture not supported
```
https://ci.status.im/job/status-react/job/combined/job/mobile-ios/4450/console

<blockquote><div><strong><a href="https://ci.status.im/job/status-react/job/combined/job/mobile-ios/4450/console">status-react » combined » mobile-ios #4450 Console [Jenkins]</a></strong></div></blockquote>

**Comment by jakubgs** [comment]
Feel free to force-push to the second commit in this PR Igor.

**Comment by mandrigin** [comment]
it works on my machine™ now

**Comment by jakubgs** [comment]
I've temporarily set the agent to use to `macos-03` since it seems to be the only one that works with `sdk` set to `iphonesimulator11.3`. We'll have to figure out what is different about that one.

**Comment by jakubgs** [comment]
Success! Here is what was uploaded to SauceLabs:
```json
{
    "filename":"6785.app.zip",
    "size":22056872,
    "md5":"4bb4637d2048c5cfce4945a3037fb6bc",
    "etag":"316e0adccbf7e03a0df2638ccb22e402",
    "username":"****"
}
```
Can you check out if it works @hetvart ?

**Comment by jakubgs** [comment]
Glorious success:
```json
{              
  "username": "****",
  "filename": "6785.app.zip",
  "size": 21820623,
  "md5": "be5f14ca10e37ddc9f1c6399de2b988e",
  "etag": "24094c5b7630332aad2005ee8dde21e8"
}
```
@hetvart, please check if this version is usable by SauceLabs.

**Comment by hetvart** [comment]
> 6785.app.zip

@jakubgs cc @mandrigin 
yes, it works (although it hangs periodically and I do not get a response from SauceLabs VM, looks like the cloud issue). the only question I have is whether it's possible to install the app under different simulator/ios configuration, e.g. iPhone Simulator 8 Plus/iOS 11.1 ?

**Review by pedropombeiro — APPROVED** [review]

**Comment by mandrigin** [comment]
@hetvart I have no idea :(

**Review by mandrigin — APPROVED** [review]

**Pull request merged**